src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +141,144 @@ class IndexLogEntryTest extends SparkFunSuite {
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+      val d1 = Files.createTempDirectory(dir, "d1")
+      val d1Path = PathUtils.makeAbsolute(d1.toString)
+      Files.createTempFile(d1, "f4", "")
+
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val content = Content.fromDirectory(dirPath)
+
+      val expected =
+        (fs.listStatus(dirPath).filter(_.isFile) ++ fs.listStatus(d1Path)).map(_.getPath).toSet
+
+      val actual = content.files.toSet
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test fromPath api creates the correct Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      // Create expected Content object.
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val fileInfos = fs.listStatus(dirPath).map(FileInfo(_))
+      val leafDir = Directory(dirPath.getName, fileInfos)
+      val expected = {
+        val rooDirectory = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir) {
+          (accum, name) =>
+            Directory(name, Seq(), Seq(accum))
+        }
+
+        Content(rooDirectory, NoOpFingerprint())
+      }
+
+      // Create actual Content object.
+      val actual = Content.fromDirectory(dirPath)
+
+      // Compare.
+      assert(actual.equals(expected))
+    }
+  }
+
+  test("Test fromLeafFiles api creates the correct Content object.") {
+    withTempPath { p =>
+      // Prepare some files and directories.
+      Files.createDirectories(Paths.get(p.getAbsolutePath))
+      val dir = Files.createDirectories(Paths.get(p.getAbsolutePath))
+      Files.createTempFile(dir, "f1", "")
+      Files.createTempFile(dir, "f2", "")
+      Files.createTempFile(dir, "f3", "")
+
+      // Create expected Content object.
+      val dirPath = PathUtils.makeAbsolute(dir.toString)
+      val fs = dirPath.getFileSystem(new Configuration)
+      val fileStatuses = fs.listStatus(dirPath)
+      val fileInfos = fileStatuses.map(FileInfo(_))
+      val leafDir = Directory(dirPath.getName, fileInfos)
+
+      val expected = {
+        val rooDirectory = TestUtils.splitPath(dirPath.getParent).foldLeft(leafDir) {
+          (accum, name) =>
+            Directory(name, Seq(), Seq(accum))
+        }
+
+        Content(rooDirectory, NoOpFingerprint())
+      }
+
+      // Create actual Content object.
+      val actual = Content.fromLeafFiles(fileStatuses)
```

Comment:
Can you add more tests for this API? Files in different dirs, files in root dir (which has a special handling), etc.

Also, I think it's better to handcraft expected and actual to make testing easier. The only thing that prevents this is `isDirectory` in `require` right? If so, can we control it by a flag and hand craft actual/expected test inputs/outputs?

Reply:
Handcrafting problem: https://github.com/microsoft/hyperspace/pull/139#discussion_r482608870

I have added other tests except the file in root dir. Since this test will also run on build machines, i am not sure of adding files directly in the root folder. Could you please ack here once and I will add the test if you are ok with this. Just want a confirmation.
Thanks. @imback82

Reply:
You don't need to create actual files to test this no, as I mentioned above?